tests/andromede/test_electrolyzer_n_inputs.py

```diff
@@ -0,0 +1,301 @@
+    DECOMPOSE_1_FLOW_INTO_2_FLOW,
+    TWO_INPUTS_CONVERTOR_MODEL,
+)
+from andromede.simulation import TimeBlock, build_problem
+from andromede.study import (
+    ConstantData,
+    DataBase,
+    Network,
+    Node,
+    PortRef,
+    create_component,
+)
+
+
+def test_electrolyzer_n_inputs_1():
+    """
+    Test with an electrolyzer for each inputs
+    """
+    elec_node_1 = Node(model=NODE_BALANCE_MODEL, id="e1")
+    electric_prod_1 = create_component(model=GENERATOR_MODEL, id="ep1")
+    electrolyzer1 = create_component(model=CONVERTOR_MODEL, id="ez1")
+
+    elec_node_2 = Node(model=NODE_BALANCE_MODEL, id="e2")
+    electric_prod_2 = create_component(model=GENERATOR_MODEL, id="ep2")
+    electrolyzer2 = create_component(model=CONVERTOR_MODEL, id="ez2")
+
+    gaz_node = Node(model=NODE_BALANCE_MODEL, id="g")
+    gaz_prod = create_component(model=GENERATOR_MODEL, id="gp")
+    gaz_demand = create_component(model=DEMAND_MODEL, id="gd")
+
+    database = DataBase()
+
+    database.add_data("ep1", "p_max", ConstantData(100))
+    database.add_data("ep1", "cost", ConstantData(30))
+    database.add_data("ez1", "alpha", ConstantData(0.7))
+
+    database.add_data("ep2", "p_max", ConstantData(100))
+    database.add_data("ep2", "cost", ConstantData(30))
+    database.add_data("ez2", "alpha", ConstantData(0.7))
+
+    database.add_data("gd", "demand", ConstantData(70))
+    database.add_data("gp", "p_max", ConstantData(10))
+    database.add_data("gp", "cost", ConstantData(40))
+
+    network = Network("test")
+    network.add_node(elec_node_1)
+    network.add_component(electric_prod_1)
+    network.add_component(electrolyzer1)
+    network.add_node(elec_node_2)
+    network.add_component(electric_prod_2)
+    network.add_component(electrolyzer2)
+    network.add_node(gaz_node)
+    network.add_component(gaz_prod)
+    network.add_component(gaz_demand)
+
+    network.connect(
+        PortRef(electric_prod_1, "balance_port"), PortRef(elec_node_1, "balance_port")
+    )
+    network.connect(
+        PortRef(elec_node_1, "balance_port"), PortRef(electrolyzer1, "FlowDI")
+    )
+    network.connect(PortRef(electrolyzer1, "FlowDO"), PortRef(gaz_node, "balance_port"))
+    network.connect(
+        PortRef(electric_prod_2, "balance_port"), PortRef(elec_node_2, "balance_port")
+    )
+    network.connect(
+        PortRef(elec_node_2, "balance_port"), PortRef(electrolyzer2, "FlowDI")
+    )
+    network.connect(PortRef(electrolyzer2, "FlowDO"), PortRef(gaz_node, "balance_port"))
+    network.connect(
+        PortRef(gaz_node, "balance_port"), PortRef(gaz_demand, "balance_port")
+    )
+    network.connect(
+        PortRef(gaz_prod, "balance_port"), PortRef(gaz_node, "balance_port")
+    )
+
+    scenarios = 1
+    problem = build_problem(network, database, TimeBlock(1, [0]), scenarios)
+    status = problem.solver.Solve()
+
+    assert status == problem.solver.OPTIMAL
+    assert problem.solver.Objective().Value() == 2971.4285714285716
```

Comment:
You should also assert the values of the solution (i.e. prod with both electrolyzer and so on), however as you use a symmetric test case, you may get different solutions depending on the solver you use, hard to test...
-> Maybe change costs or efficiency between electrolyzer to get a unique solution

Reply:
This applies to all following test cases